docs(weave): add custom runtimes guide (#2996)
wandb/docs: weave/guides/tools/custom-runtimes.mdx

---
title: "Add custom runtimes"
description: "Connect OpenAI-compatible inference endpoints to a Weave project as custom runtimes, and manage them from the UI or the Python and TypeScript SDKs."
keywords: ["custom runtimes", "custom providers", "OpenAI-compatible endpoint", "register_custom_runtime", "LLM playground", "LLM judge"]
---

> A custom runtime connects an OpenAI-compatible inference endpoint to a W&B Weave project. After you register a custom runtime, its models appear in model selectors across Weave, so you and your team can chat with the endpoint in the Playground, evaluate it in the Evaluation Playground, or use it as an LLM judge, without anyone writing integration code.

Reviewer: This sentence feels a bit long, confusing. Split into two?

Reviewer: E.g. maybe something like?

```suggestion
Register a custom runtime to connect an OpenAI-compatible inference endpoint to a W&B Weave project. After registration, the endpoint’s models appear in model selectors across Weave. You and your team can then use them in the Playground, Evaluation Playground, and LLM-as-a-judge workflows without writing integration code.
```

> A custom runtime connects an OpenAI-compatible inference endpoint to a W&B Weave project. After you register a custom runtime, its models appear in model selectors across Weave, so you and your team can chat with the endpoint in the Playground, evaluate it in the Evaluation Playground, or use it as an LLM judge, without anyone writing integration code.

Reviewer 2: ```suggestion
A custom runtime connects an OpenAI-compatible inference endpoint to a W&B Weave project. After you register a custom runtime, its models appear in model selectors across Weave. Your team can then chat with the endpoint in the Playground, evaluate it in the Evaluation Playground, or use it as an LLM judge, all without writing integration code.
```

Use a custom runtime to connect any of the following:

- An agent or model prototype your team serves behind an OpenAI-compatible API.
- A self-hosted or local model (for example, one served by Ollama or vLLM).

> - A provider or model version that Weave doesn't ship built-in support for.

Reviewer 2: ```suggestion
- A provider or model version that Weave doesn't have built-in support for.
```
